hypershift-operator/controllers/nodepool/nodepool_controller_test.go

```diff
@@ -858,10 +863,34 @@ func TestMachineTemplateBuilders(t *testing.T) {
 		},
 	}
 
+	if preCreateMachineTemplate {
```

Comment:
fwiw usually we do test cases the likes of
```
	testCases := []struct {
		name     string
		existing
		expected
	}{
```

then we pass existing to the withObjects function of the fake client. See e.g TestGetConfig.

But this works as well, we can follow up on it later.

Reply:
also the func don't need to be exported